diff --git a/googlesql/base/BUILD b/googlesql/base/BUILD
--- a/googlesql/base/BUILD
+++ b/googlesql/base/BUILD
@@ -345,6 +345,7 @@ cc_library(
     deps = [
         ":requires",
         "@com_google_absl//absl/base:core_headers",
+        "@com_google_absl//absl/base:throw_delegate",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:cord",
     ],
diff --git a/googlesql/base/associative_view_internal.h b/googlesql/base/associative_view_internal.h
--- a/googlesql/base/associative_view_internal.h
+++ b/googlesql/base/associative_view_internal.h
@@ -29,6 +29,7 @@
 
 #include "absl/base/attributes.h"
 #include "absl/base/optimization.h"
+#include "absl/base/throw_delegate.h"
 #include "absl/strings/cord.h"
 #include "absl/strings/string_view.h"
 #include "googlesql/base/requires.h"
@@ -155,7 +156,7 @@ template <typename mapped_type, typename View, typename Key>
 const mapped_type& AtFunction(const View& v, const Key& key) {
   auto it = v.find(key);
   if (ABSL_PREDICT_FALSE(it == v.end())) {
-    absl::base_internal::ThrowStdOutOfRange(
+    absl::ThrowStdOutOfRange(
       "MapView::at failed bounds check");
   }
   return it->second;
